tests/acceptance/useragent/useragentsuite.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable quote-props */
+import { UNIVERSAL_PAGE } from '../constants';
+import { ClientFunction } from 'testcafe';
+import { isChrome, isSafari, isMicrosoftEdge } from '../../../src/core/utils/useragent';
+
+fixture`User-Agent sniffing`
+  .page`${UNIVERSAL_PAGE}`;
+
+test('User-Agent sniffing works as expected', async t => {
+  const browserNameToSniffingMethod = {
+    'Chrome': isChrome,
+    'Safari': isSafari,
+    'Microsoft Edge': isMicrosoftEdge,
+    'Firefox': () => t.browser.name === 'Firefox'
+  };
+  if (!(t.browser.name in browserNameToSniffingMethod)) {
+    throw new Error(
+      `Unexpected testcafe browser "${t.browser.name}".\n` +
+      `Expected one of ${Object.keys(browserNameToSniffingMethod).join(', ')}`);
+  }
+  for (const [browserToTest, sniffer] of Object.entries(browserNameToSniffingMethod)) {
+    const [userAgent, userAgentData] = await ClientFunction(() => {
+      return [window.navigator.userAgent, window.navigator.userAgentData];
+    })();
+    global.navigator = { userAgent, userAgentData };
```

Comment:
can we not directly access window.navigator.[userAgent|userAgentData] in isSafari or isChrome? or is navigator different from window.navigator

Reply:
We can't run isSafari directly in the browser through testcafe (I tried that initially), ClientFunction doesn't let you pass arbitrary code in

Reply:
like this is what happens
```js
function someFunctionIWantInTheBrowser() {
  console.log('blah');
}
await ClientFunction(() => {
  someFunctionIWantInTheBrowser()
  // Error is thrown "someFunctionIWantInTheBrowser is not defined"
})();
```